src/main/java/org/folio/search/controller/SearchConsortiumController.java

```diff
@@ -82,11 +93,68 @@ public ResponseEntity<ConsortiumLocationCollection> getConsortiumLocations(Strin
       .totalRecords(result.getTotalRecords()));
   }
 
-  private void checkAllowance(String tenantHeader) {
+  @Override
+  public ResponseEntity<ConsortiumHolding> getConsortiumHolding(UUID id, String tenantHeader) {
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var holdingId = id.toString();
+    var query = "holdings.id=" + holdingId;
+    var searchRequest = CqlSearchRequest.of(Instance.class, tenant, query, 1, 0, true, false, true);
+
+    var result = searchService.getConsortiumHolding(id.toString(), searchRequest);
+    return ResponseEntity.ok(result);
+  }
+
+  @Override
+  public ResponseEntity<ConsortiumItem> getConsortiumItem(UUID itemId, String tenantHeader) {
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var query = "items.id=" + itemId.toString();
+    var searchRequest = CqlSearchRequest.of(Instance.class, tenant, query, 1, 0, true, false, true);
+
+    var result = searchService.getConsortiumItem(itemId.toString(), searchRequest);
+    return ResponseEntity.ok(result);
+  }
+
+  @Override
+  public ResponseEntity<ConsortiumHoldingCollection> fetchConsortiumBatchHoldings(String tenantHeader,
+                                                                                  BatchIdsDto batchIdsDto) {
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var holdingIds = batchIdsDto.getIds().stream().map(UUID::toString).collect(Collectors.toSet());
+    var query = batchIdsDto.getIds().stream()
+      .map(UUID::toString)
+      .map("holdings.id = %s"::formatted)
+      .collect(Collectors.joining(" or "));
+    var searchRequest = CqlSearchRequest.of(Instance.class, tenant, query, holdingIds.size(), 0, true, false, true);
+
+    var result = searchService.fetchConsortiumBatchHoldings(searchRequest, holdingIds);
+    return ResponseEntity.ok(result);
+  }
+
+  @Override
+  public ResponseEntity<ConsortiumItemCollection> fetchConsortiumBatchItems(String tenantHeader,
+                                                                            BatchIdsDto batchIdsDto) {
+    if (batchIdsDto.getIds().isEmpty()) {
+      return ResponseEntity
+        .ok(new ConsortiumItemCollection());
+    }
+
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var itemIds = batchIdsDto.getIds().stream().map(UUID::toString).collect(Collectors.toSet());
+    var query = batchIdsDto.getIds().stream()
+      .map(UUID::toString)
+      .map("items.id = %s"::formatted)
+      .collect(Collectors.joining(" or "));
+    var searchRequest = CqlSearchRequest.of(Instance.class, tenant, query, itemIds.size(), 0, true, false, true);
```

Comment:
looks like lines 142-146 could be extracted to some method and reused also in batch holdings endpoint for building a query and request